frontend/src/modules/Traces/TraceFilter.tsx

```diff
@@ -396,6 +429,20 @@ const _TraceFilter = (props: TraceFilterProps) => {
 					/>
 				</FormItem>
 
+				<FormItem name="spanKind">
+					<Select
+						showSearch
+						style={{ width: 180 }}
+						onChange={handleChangeSpanKind}
+						placeholder="Select Span Kind"
+						allowClear
+					>
+						{spanKindList && spanKindList.map(spanKind => (
```

Comment:
Since spankindlist already exists and is constant we don't need to have this condition.

Reply:
Hi @jyash97, I think we can still keep that check if `spanKindList` does exist or not. It gives an extra layer of check and if somehow `spanKindList` get deleted from the code, this check will prevent from throwing an error to the frontend. Let me know if you were thinking from some other perspective.

Reply:
If someone removes the variable it should break the app since that is not expected. We should add the checks when we deal with state variables.

Reply:
Yeah, I agree with you that the app will break if someone removes the variable and that's the reason I'm checking if the variable does exist or not. And also I don't think that there is a specific convention in React that if we are using state then only we should add checks. It just a check, neither it's creating an extra memory allocation nor hampering the execution. It for a good reason!!

Reply:
> if someone removes the variable and that's the reason I'm checking if the variable does exist or not.

I meant to say the error thrown would be in development mode.

Also you may have to update the check to:

```
Array.isArray(list)
```

Reply:
@reachtokish @jyash97 We can pick this up in a fresh issue. Merging the PR for now.